src/src/ESPEasyCore/ESPEasyRules.cpp

```diff
@@ -1307,6 +1308,34 @@ bool timeStringToSeconds(const String& tBuf, int& time_seconds, String& timeStri
   return validTime;
 }
 
+// Balance the count of parentheses (aka round braces) by adding the missing left or right parentheses, if any
+// Returns the number of added parentheses, < 0 is left parentheses added, > 0 is right parentheses added
+int balanceParentheses(String& string) {
+  int left = 0;
+  int right = 0;
+  for (unsigned int i = 0; i < string.length(); i++) {
+    switch (string[i]) {
+      case '(':
+        left++;
+        break;
+      case ')':
+        right++;
+        break;
+    }
+  }
+  if (left > right) {
+    for (int i = 0; i < left - right; i++) {
```

Comment:
May also benefit from a call to reserve
`string.reserve(string.length() + (right - left));`